src/software/gui/drawing/draw_functions.h

```diff
@@ -3,17 +3,23 @@
 #include <QtWidgets/QGraphicsScene>
 #include <functional>
 
-// This class is used to represent a "draw function", which is a function
-// provided to the visualizer that tells is how to draw things.
-// All functionality is implemented in this class so
-// subclasses only need to call the superconstructor.
-//
-// This class is made abstract by making the desctructor pure virtual,
-// but providing an implementation. The subclasses will have a default
-// destructor automatically generated which will implement the virtual
-// destructor. All together this lets the base class be pure virtual
-// without needing an extra pure-virtual "dummy" function
-// See https://stackoverflow.com/a/4641108
+/**
+ * This class is used to represent a "draw function", which is a function
+ * provided to various GUI components that tells them how to draw things.
+ *
+ * Draw functions exist because certain objects we would normally like to
+ * draw (like the AI) are non-copyable, so they can't be sent over
+ * the Observer system. We'd like to keep GUI / drawing-specific code
+ * out of our logic as much as possible, so DrawFunctions are the
+ * solution to that.
+ *
+ * DrawFunctions can capture any data that supports copying and
+ * "store" it for later. That way the entire function can be copied
+ * and passed around the system, until some consumer calls it to
+ * draw its contents. This way we can hide the details of "how"
+ * something is drawn, but still pass around complex DrawFunctions
+ * in order to draw non-copyable objects.
```

Comment:
There seems to be some contradiction about things being copyable and non-copyable here?

Reply:
clarified, let me know if that's better.